### PR TITLE
[vm] Add assertion for discarded transaction.

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Add an extra assertion in Move functional test to enforce all discarded transaction should have an empty writeset.
 
### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: language/functional-tests/src/evaluator.rs [hunk 1]
```diff
@@ -343,7 +343,10 @@ fn run_transaction(
                     Err(ErrorKind::VMExecutionFailure(output).into())
                 }
             }
-            TransactionStatus::Discard(_) => Err(ErrorKind::DiscardedTransaction(output).into()),
+            TransactionStatus::Discard(_) => {
+                assert!(output.write_set().is_empty());
```
Comment (hermanventer) [inline language/functional-tests/src/evaluator.rs]: I would certainly appreciate it if you could rather use checked_verify! from mirai-annotations v1.6.0. It generates the exact same code as present when MIRAI not the compiler, but gives MIRAI more information than assert does because it distinguishes the path condition from the verification condition.

File: language/functional-tests/Cargo.toml [hunk 2]
```diff
@@ -24,3 +24,4 @@ thiserror = "1.0"
 aho-corasick = "0.7.6"
 termcolor = "1.0.5"
 datatest-stable = { path = "../../common/datatest-stable", version = "0.1.0" }
+mirai-annotations = "1.4.0"
```
Comment (hermanventer) [inline language/functional-tests/Cargo.toml]: The latest version is 1.6.0.

Comment (runtian-zhou) [reply language/functional-tests/Cargo.toml]: Thanks for the catch!
